integrations/virtiofs/src/filesystem.rs

```diff
@@ -35,14 +40,94 @@ const BUFFER_HEADER_SIZE: u32 = 256;
 /// The maximum length of the data part of the message, used for read/write data.
 const MAX_BUFFER_SIZE: u32 = 1 << 20;
 
+/// FileType represents the type of the opened file.
+enum FileType {
+    Dir,
+    File,
+    Unknown,
+}
+
+/// FileKey represents the key assigned to the opened file.
+#[derive(Clone, Copy)]
+struct FileKey(usize);
+
+/// OpenedFile represents file that opened in memory.
+#[allow(dead_code)]
```

Comment:
Hi, from my perspective, adding `dead_code` is not ideal. If they are useful, please consider using `pub use` to include them.